### DISCUSSION TITLE
Vue (with Laravel Inertia) route linking to the controller

### DISCUSSION THREAD

**Discussion by LudovicLemarinel** [vuejs/language-tools#4834] [Q&A]
Hi

I'm mainly using Vue with laravel and Inertia.
Laravel exposes named routes (such as "posts.index"), and those routes names can be used in vue with the `route('posts.index')`shortcut.

In phpstorm, those named routes are resolved and phpstorm allows you to click on it to jump to the corresponding controller. You even have autocompletion on routes names ;)

But phpstorm has big issues when the project becomes to be a bit too big and my (despite brand-new M3 cpu) macbook struggles ... So I'm trying to switch to vscode and I miss those little things ...

Any ways to enable those links and autocompletion ?

**Comment by RayGuo-ergou** [comment]
I had worked with laravel livewire in the past, as far as I know. Laravel support in other text editors are very limited. Which makes sense because `phpstorm` is a php specific IDE and it's a paid product. You can search for plugins on vscode market see if any plugin can suit your need.

It should be easier to resolve in a file system level than language server level e.g. go through the route files and emit to the auto completion list. Even in language server level, I reckon it should be addressed via php lsp. 

Just a hint/FYI:
I am using neovim and this [plugin](https://github.com/ricardoramirezr/blade-nav.nvim) to achieve "go to definition" and "auto completion" for routes. If you cannot find a plugin in vscode market probably this can be an inspiration for you to build a simple plugin to do the same thing. :)

**Reply by LudovicLemarinel** [reply]
I think that plugins that can resolve routes (perhaps phpstorm also) are using the "laravel route:list" command since it returns routes that can be defined many ways (route attribute package, routes/web.php file..., other packages that subscribe routes)

I'll wait for the official laravel plugin to come (should arrive october), and see if they included vuejs support for route linking...
